### ISSUE TITLE
Address already in use error with mongo fixture

### ISSUE THREAD

**Issue by bmoscon** [manahl/pytest-plugins#34]
see: https://travis-ci.org/manahl/arctic/jobs/171243541

```
2016-10-27 23:03:05,719 DEBUG pytest_shutil.workspace 
2016-10-27 23:03:05,719 DEBUG pytest_shutil.workspace =======================================================
2016-10-27 23:03:05,719 DEBUG pytest_shutil.workspace pytest_shutil using workspace /tmp/mongo/tmpk8rohnql
2016-10-27 23:03:05,719 DEBUG pytest_shutil.workspace This workspace will delete itself on teardown
2016-10-27 23:03:05,720 DEBUG pytest_shutil.workspace =======================================================
2016-10-27 23:03:05,720 DEBUG pytest_shutil.workspace 
2016-10-27 23:03:05,720 DEBUG pytest_server_fixtures.base Starting Server on host testing-worker-linux-docker-b7c2ec4f-3396-linux-1 port 38908
2016-10-27 23:03:05,759 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.758 [initandlisten] MongoDB starting : pid=5338 port=38908 dbpath=/tmp/mongo/tmpk8rohnql 64-bit host=testing-worker-linux-docker-b7c2ec4f-3396-linux-1
2016-10-27 23:03:05,759 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten]
2016-10-27 23:03:05,759 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] ** WARNING: You are running on a NUMA machine.
2016-10-27 23:03:05,759 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] **          We suggest launching mongod like this to avoid performance problems:
2016-10-27 23:03:05,759 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] **              numactl --interleave=all mongod [other options]
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten]
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] db version v2.4.12
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] git version: 09917767b116f4ff1c0eadda1e8bc5db30828500
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] build info: Linux ip-10-142-184-243 2.6.21.7-2.ec2.v1.2.fc8xen #1 SMP Fri Nov 20 17:48:28 EST 2009 x86_64 BOOST_LIB_VERSION=1_49
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] allocator: tcmalloc
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [initandlisten] options: { bind_ip: "testing-worker-linux-docker-b7c2ec4f-3396-linux-1", dbpath: "/tmp/mongo/tmpk8rohnql", nohttpinterface: true, nojournal: true, nounixsocket: true, nssize: 1, port: 38908, quiet: true, smallfiles: true, syncdelay: 0.0 }
2016-10-27 23:03:05,760 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.759 [DataFileSync] warning: --syncdelay 0 is not recommended and can have strange performance
2016-10-27 23:03:05,761 DEBUG pytest_server_fixtures.base Running server: /usr/bin/mongod --bind_ip=testing-worker-linux-docker-b7c2ec4f-3396-linux-1 --port=38908 --dbpath=/tmp/mongo/tmpk8rohnql --nounixsocket --smallfiles --syncdelay 0 --nohttpinterface --nssize=1 --nojournal --quiet
2016-10-27 23:03:05,762 DEBUG pytest_server_fixtures.base sleeping for 0.1 before retrying (1 of 28)
2016-10-27 23:03:05,767 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.765 [initandlisten] allocating new ns file /tmp/mongo/tmpk8rohnql/local.ns, filling with zeroes...
2016-10-27 23:03:05,762 DEBUG pytest_server_fixtures.base CWD: /home/travis/build/manahl/arctic
2016-10-27 23:03:05,768 DEBUG pytest_server_fixtures.base waited 0:00:00.007156 for server to start successfully
2016-10-27 23:03:05,768 DEBUG pytest_server_fixtures.base Server now awake
2016-10-27 23:03:05,769 INFO arctic.fixtures.arctic arctic.fixtures: arctic init()
2016-10-27 23:03:05,772 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.772 [FileAllocator] allocating new datafile /tmp/mongo/tmpk8rohnql/local.0, filling with zeroes...
2016-10-27 23:03:05,772 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.772 [FileAllocator] creating directory /tmp/mongo/tmpk8rohnql/_tmp
2016-10-27 23:03:05,773 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.773 [FileAllocator] done allocating datafile /tmp/mongo/tmpk8rohnql/local.0, size: 16MB,  took 0 secs
2016-10-27 23:03:05,774 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] ERROR: listen(): bind() failed errno:98 Address already in use for socket: 172.17.4.90:38908
2016-10-27 23:03:05,774 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] ERROR:   addr already in use
2016-10-27 23:03:05,774 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] now exiting
2016-10-27 23:03:05,774 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 dbexit:
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] shutdown: going to close listening sockets...
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] shutdown: going to flush diaglog...
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] shutdown: going to close sockets...
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] shutdown: waiting for fs preallocator...
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] shutdown: closing all files...
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] closeAllFiles() finished
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 [initandlisten] shutdown: removing fs lock...
2016-10-27 23:03:05,775 DEBUG pytest_server_fixtures.base Thu Oct 27 23:03:05.774 dbexit: really exiting now
```

**Comment by eeaston** [comment]
That's very unusual, it should be picking a guaranteed unused port as assigned by the kernel. Is this happening consistently for you?

**Comment by bmoscon** [comment]
Yep, look at the build history on my unpin branch in Arctic

**Comment by jamesblackburn** [comment]
As discussed with @bmoscon we're missing a few patches here.

**Comment by bmoscon** [comment]
take a look at #35

**Comment by bmoscon** [comment]
fixed with #35
